dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/ContentletCacheImpl.java

```diff
@@ -133,6 +149,20 @@ public void remove(String key){
     	
     	String myKey = primaryGroup + key;
     	String metadataKey = metadataGroup + key;
+    	com.dotmarketing.portlets.contentlet.model.Contentlet content = null;
+    	try {
+    		content = (com.dotmarketing.portlets.contentlet.model.Contentlet)cache.get(key,primaryGroup);
+    		try {
+				if(content != null && content.isVanityUrl()){
```

Comment:
we might need a higher level vanityurl api call, something like `invalidateVanityUrl(Contentlet con)`